src/app/streckengrafik/services/sg-1-load-trainrun-item.service.ts

```diff
@@ -624,33 +669,56 @@ export class Sg1LoadTrainrunItemService implements OnDestroy {
       }
     });
 
-    return new TrainrunItem(
-      trainrun.getId(),
-      trainrun.getFrequency(),
-      trainrun.getFrequencyOffset(),
-      trainrunStartTime,
-      trainrunEndTime,
-      trainrun.getTitle(),
-      trainrun.getCategoryShortName(),
-      trainrun.getCategoryColorRef(),
-      pathItems,
-    );
+    return {
+      trainrunItem: new TrainrunItem(
+        trainrun.getId(),
+        trainrun.getFrequency(),
+        trainrun.getFrequencyOffset(),
+        trainrunStartTime,
+        trainrunEndTime,
+        trainrun.getTitle(),
+        trainrun.getCategoryShortName(),
+        trainrun.getCategoryColorRef(),
+        pathItems
+      ),
+      visitedTrainrunSections: visitedTrainrunSections
+    };
   }
 
   public loadTrainrunItems(
     templateTrainrunItem: TrainrunItem,
   ): TrainrunItem[] {
+    // Extract for all trainruns the sections which are part of the
+    // template path (along which the graphical timetable is projected) and it does
+    // support current for those trains with partial cancellations.
     const trainrunItems: TrainrunItem[] = [];
     if (this.trainruns) {
       this.trainruns.forEach((trainrun: Trainrun) => {
         if (this.filterService.filterTrainrun(trainrun)) {
-          const trainrunItem = this.loadTrainrunItem(trainrun, false);
-          this.sortTrainrunItemAndRotateAlongTemplatePath(
-            trainrun,
-            trainrunItem,
-            templateTrainrunItem,
-          );
-          trainrunItems.push(trainrunItem);
+          // get all trainrun section for given trainrun
+          let alltrainrunsections = this.trainrunSectionService
+            .getAllTrainrunSectionsForTrainrun(trainrun.getId());
+
+          // As long not all trainrun section are visited (process) continue
+          // this part of code supports partial cancelatlations, e.g. trainrun runs from
```

Comment:
```suggestion
          // this part of code supports partial cancellations, e.g., trainrun runs from
```
see
https://www.merriam-webster.com/thesaurus/cancellation
and
https://www.merriam-webster.com/dictionary/e.g.#examples